packages/react-day-picker/src/components/YearsDropdown/YearsDropdown.tsx

```diff
@@ -64,6 +65,10 @@ export function YearsDropdown(props: YearsDropdownProps): JSX.Element {
       onChange={handleChange}
       value={displayMonth.getFullYear()}
       caption={formatYearCaption(displayMonth, { locale })}
+      options={years.map((m) => ({
+        value: m.getMonth(),
+        name: stringFormatYearCaption(m, { locale })
```

Comment:
Why can't we use `formatters.formatMonthCaption`? Or do you meant to add `formatMonthOption` or something?

Reply:
The issue is that `formatMonthCaption` has the return type of a JSX element. I created this because I did not want to introduce a breaking change. Changing it to a `string` return type would however be trivial, as it already returns a string - just has the wrong type information for this use. What option do you think is better?

Reply:
> The issue is that `formatMonthCaption` has the return type of a React node. I created this because I did not want to introduce a breaking change.

Oh I see it now, thanks for the explanation. There are many ways for skipping a breaking change (e.g. we could add a new `formatDropdownOption: () => string` function to the formatters).